File: Webpage/server/webpage.server/webpage.server/src/main/java/abbruzzese/webpage/webpage/server/webpage/Mappers/ProductMapper.java
package abbruzzese.webpage.webpage.server.webpage.Mappers;

import org.springframework.stereotype.Component;
import abbruzzese.webpage.webpage.server.webpage.DTO.DTOProduct;
import abbruzzese.webpage.webpage.server.webpage.entities.Product;

@Component(value="ProductMapper")
public class ProductMapper {

    public Product DTOtoProduct(DTOProduct dtoprod){
    Product prod=new Product();
    prod.setNombre(dtoprod.getNombre());
    prod.setMarca(dtoprod.getMarca());
    prod.setCategoria(dtoprod.getCategoria());
    prod.setFecha(dtoprod.getFecha());
    prod.setImageproduct(dtoprod.getImageproduct());
    return prod;

    } 

    public DTOProduct ProducttoDTO(Product prod){

        DTOProduct dtoprod=new DTOProduct();
        dtoprod.setNombre(prod.getNombre());
        dtoprod.setMarca(prod.getMarca());
        dtoprod.setCategoria(prod.getCategoria());
        dtoprod.setFecha(prod.getFecha());
        dtoprod.setImageproduct(prod.getImageproduct());
        return dtoprod;
        
    }
}

File: PHP_API/MAPPER/ProductMapper.php
<?php 

namespace App\Mapper;

use App\Entity\Product;
use App\DTO\DTOProduct;

class ProductMapper{

    public static function ProducttoDTO(Product $product){
       return   new DTOProduct(
            $product->getNombre(),
            $product->getMarca(),
            $product->getCategoria(),
            $product->getImageproduct(),
            $product->getLocaldate(),
        );

    }

    public static function DTOtoProduct(DTOProduct $DTOproduct){
       return new Product(
            $DTOproduct->getNombre(),
            $DTOproduct->getMarca(),
            $DTOproduct->getCategoria(),
            $DTOproduct->getImageproduct(),
        );
    }

}

?>